### PR TITLE
Enable Multi-tenancy for Weaviate

### PR DESCRIPTION
**Pull request by khajavi**
- **Description:** Added support for multi-tenancy when creating a schema and taking tenant name to index and retrieve data.
  - **Dependencies:** nothing
  - **Tag maintainer:** @hwchase17 
  - **Twitter handle:** https://twitter.com/khajavi

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/weaviate.py [hunk 1]
```diff
@@ -509,7 +521,6 @@ def from_texts(
             embedding=embedding,
             attributes=attributes,
             relevance_score_fn=relevance_score_fn,
-            by_text=by_text,
```
Comment (hwchase17) [inline libs/langchain/langchain/vectorstores/weaviate.py:512]: why is this removed?

File: libs/langchain/langchain/vectorstores/weaviate.py [hunk 2]
```diff
@@ -494,6 +505,7 @@ def from_texts(
                     "uuid": _id,
                     "data_object": data_properties,
                     "class_name": index_name,
+                    "tenant": tenant,
```
Comment (hwchase17) [inline libs/langchain/langchain/vectorstores/weaviate.py:508]: this still works if it is None?

Comment (khajavi) [reply libs/langchain/langchain/vectorstores/weaviate.py:508]: reverted that change!

File: libs/langchain/langchain/vectorstores/weaviate.py (hunk 1 above)
Comment (khajavi) [reply libs/langchain/langchain/vectorstores/weaviate.py:512]: reverted that change.

File: libs/langchain/langchain/vectorstores/weaviate.py (hunk 2 above)
Comment (khajavi) [reply libs/langchain/langchain/vectorstores/weaviate.py]: fixed that.

**Comment by khajavi** [comment]
@hwchase17 
Thank you for the review! Fixed those comments.
I have the problem running `make format` and `make lint` on NixOs:

```
make lint                                                                                                                          
poetry run ruff docs templates cookbook                                                                                                                                                                       
Command not found: ruff                                                                                                                                                                                       
make: *** [Makefile:45: lint] Error 1 
```

```
(.venv) milad@nixos ~/s/p/langchain (enable-weaviate-multi-tenancy)> ./.venv/bin/ruff --help
fish: Job 1, './.venv/bin/ruff --help' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by khajavi** [comment]
Thank you for your help! Everything is now green.

**Comment by baskaryan** [comment]
believe you need to specify tenant names when doing queries as well, `client.query.get("...").with_tenant("...").do()`

cc @hsm207

**Comment by hsm207** [comment]
> believe you need to specify tenant names when doing queries as well, `client.query.get("...", ["...",]).with_tenant("...").do()`

Yes, you do need to do that.

This PR is incomplete without it. I recommend writing tests to verify the CRUD operations work in a multi tenant setup.

**Comment by khajavi** [comment]
@baskaryan @hsm207 
There is no need to have such a change, there is an existing line on the repo already:
https://github.com/langchain-ai/langchain/blob/55de9d5d3c463cf40a0731aa807851a24309493a/libs/langchain/langchain/vectorstores/weaviate.py#L215

**Comment by khajavi** [comment]
fixed the conflicts and ready for any other feedback you have.

**Comment by khajavi** [comment]
Any update?

**Comment by khajavi** [comment]
@hwchase17 Can you please take a look?

**Comment by khajavi** [comment]
"Hey @hwchase17, Could you please let me know why you closed it? Also, could you review/merge it?"

**Comment by efriis** [comment]
Hey @khajavi ! This package is actually being moved out into its own repo by @hsm207 . Feel free to put up a PR over there, or @hsm207 may have already implemented it in the new version!

https://github.com/langchain-ai/langchain-weaviate

**Comment by hsm207** [comment]
fyi, support for multi tenancy is already implemented in the new repo. See langchain-ai/langchain-weaviate#25
